File: app/src/main/java/net/oliversne/volumecalculator/MainActivity.java
package net.oliversne.volumecalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Adapter;
import android.widget.AdapterView;
import android.widget.GridLayout;
import android.widget.GridView;
import android.widget.Toast;

import net.oliversne.volumecalculator.Adapters.Shapes;
import net.oliversne.volumecalculator.Models.Shape;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    //Adapter View
    GridView gridView;
    //Data Source
    ArrayList<Shape> shapeArrayList;
    //Adapter
    Shapes adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Set the Adapter View
        gridView = findViewById(R.id.grid_view);

        //Set the Data Source
        shapeArrayList = new ArrayList<>();

        Shape shape1 = new Shape(R.drawable.cube, "Cube");
        Shape shape2 = new Shape(R.drawable.cylinder, "Cylinder");
        Shape shape3 = new Shape(R.drawable.prism, "Prism");
        Shape shape4 = new Shape(R.drawable.sphere, "Sphere");

        shapeArrayList.add(shape1);
        shapeArrayList.add(shape2);
        shapeArrayList.add(shape3);
        shapeArrayList.add(shape4);

        //Set the Adapter
        adapter = new Shapes(shapeArrayList,this);
        gridView.setAdapter(adapter);
        gridView.setNumColumns(2);

        gridView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                String item = adapter.getItem(position).getShapeName();
                Intent intent = new Intent(getApplicationContext(),VolumeShapes.class);
                intent.putExtra("shapeName", item);
                startActivity(intent);
            }
        });

    }
}

File: app/src/main/java/com/adrianhelo/volumecalculator/ui/main/MainActivity.kt
package com.adrianhelo.volumecalculator.ui.main

import android.content.Intent
import android.os.Bundle
import android.widget.GridView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import com.adrianhelo.volumecalculator.ui.CylinderActivity
import com.adrianhelo.volumecalculator.ui.PrismActivity
import com.adrianhelo.volumecalculator.R
import com.adrianhelo.volumecalculator.ui.CubeActivity
import com.adrianhelo.volumecalculator.ui.SphereActivity
import com.adrianhelo.volumecalculator.ui.adapters.ShapeAdapter
import com.adrianhelo.volumecalculator.ui.models.Shape

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        // ListView
        val gridView: GridView = findViewById(R.id.grid_view)
        // Data Source
        var shape1 = Shape("Cube", R.drawable.cube)
        var shape2 = Shape("Sphere", R.drawable.sphere)
        var shape3 = Shape("Cylinder", R.drawable.cylinder)
        var shape4 = Shape("Prism", R.drawable.prism)
        val shapes = listOf<Shape>(shape1, shape2, shape3, shape4)

        // Custom Adapter
        val adapter = ShapeAdapter(this, shapes)
        //Binding Adapter
        gridView.adapter = adapter

        gridView.setOnItemClickListener{_, _, it, _ ->
            val clickedItem = adapter.getItem(it)

            if (clickedItem?.shapeName.equals("Cube")){
                val i = Intent(this, CubeActivity::class.java)
                startActivity(i)
            } else if (clickedItem?.shapeName.equals("Sphere")){
                val i = Intent(this, SphereActivity::class.java)
                startActivity(i)
            } else if (clickedItem?.shapeName.equals("Cylinder")){
                val i = Intent(this, CylinderActivity::class.java)
                startActivity(i)
            } else {
                val i = Intent(this, PrismActivity::class.java)
                startActivity(i)
            }
        }
    }
}